ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+            pszDesc = MMReturnValueFromSectionINIFile(pszRelFile, section,
+                                                      "descriptor_cat");
+            if (pszDesc)
+            {
+                pszString = CPLRecode_function(pszDesc, CPL_ENC_ISO8859_1,
+                                               CPL_ENC_UTF8);
+                free_function(pszDesc);
+
+                MM_strnzcpy(
+                    pMMBDXP->pField[nIField].FieldDescription[MM_CAT_LANGUAGE],
+                    pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+
+                if (*pMMBDXP->pField[nIField]
+                         .FieldDescription[MM_DEF_LANGUAGE] != '\0')
+                {
+                    MM_strnzcpy(pMMBDXP->pField[nIField]
+                                    .FieldDescription[MM_DEF_LANGUAGE],
+                                pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+                }
+
+                free_function(pszString);
+            }
+            else
+                *pMMBDXP->pField[nIField].FieldDescription[MM_CAT_LANGUAGE] =
+                    '\0';
+
+            // MM_SPA_LANGUAGE
+            pszDesc = MMReturnValueFromSectionINIFile(pszRelFile, section,
+                                                      "descriptor_cat");
+            if (pszDesc)
+            {
+                pszString = CPLRecode_function(pszDesc, CPL_ENC_ISO8859_1,
+                                               CPL_ENC_UTF8);
+                free_function(pszDesc);
+
+                MM_strnzcpy(
+                    pMMBDXP->pField[nIField].FieldDescription[MM_SPA_LANGUAGE],
+                    pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+
+                if (*pMMBDXP->pField[nIField]
+                         .FieldDescription[MM_DEF_LANGUAGE] != '\0')
+                {
+                    MM_strnzcpy(pMMBDXP->pField[nIField]
+                                    .FieldDescription[MM_DEF_LANGUAGE],
+                                pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+                }
+
+                free_function(pszString);
+            }
+            else
+                *pMMBDXP->pField[nIField].FieldDescription[MM_SPA_LANGUAGE] =
+                    '\0';
+        }
+    }
+
+    if (!pMMBDXP->nFields)
+    {
+        if (pMMBDXP->BytesPerRecord)
+            incoherent_record_size = TRUE;
+    }
+    else if (pMMBDXP->pField[pMMBDXP->nFields - 1].BytesPerField +
+                 pMMBDXP->pField[pMMBDXP->nFields - 1].AcumulatedBytes >
+             pMMBDXP->BytesPerRecord)
+        incoherent_record_size = TRUE;
+    if (incoherent_record_size)
+    {
+        if (n_queixes_estructura_incorrecta == 0)
+        {
+            incoherent_record_size = FALSE;
+            fseek_function(pf, offset_reintent, SEEK_SET);
+            n_queixes_estructura_incorrecta++;
+            goto reintenta_lectura_per_si_error_CreaCampBD_XP;
+        }
+    }
+
+    offset_possible = 32 + 32 * (pMMBDXP->nFields) + 1;
```

Comment:
if nFields could be > UINT32_MAX / 32, that would overflow. There should likely be some validation of nFields against a reasonable value (like 2000 ?) as soon as it is read from the file  / computed

Reply:
It's been discussed in other comment. I close that